src/fu-engine.c

```diff
@@ -3579,8 +3610,17 @@ fu_engine_add_device (FuEngine *self, FuDevice *device)
 						       NULL);
 			if (release != NULL) {
 				g_autoptr(FwupdRelease) rel = fwupd_release_new ();
-				fu_engine_set_release_from_appstream (self, rel, component, release);
-				fu_device_add_release (device, rel);
+				g_autoptr(GError) error_local = NULL;
+				if (!fu_engine_set_release_from_appstream (self,
+									   rel,
+									   component,
+									   release,
+									   &error_local)) {
+					g_warning ("failed to set release: %s",
```

Comment:
can you make this a unique string?  It makes tracking errors down later easier in my opinion.

Reply:
Sure thing.